cuda_core/tests/test_event.py

```diff
@@ -37,8 +38,13 @@ def test_timing(init_cuda, enable_timing):
         # We only want to exercise the __sub__ method, this test is not meant
         # to stress-test the CUDA driver or time.sleep().
         delay_ms = delay_seconds * 1000
-        generous_tolerance = 20
-        assert delay_ms <= elapsed_time_ms < delay_ms + generous_tolerance
+        if os.name == "nt":  # noqa: SIM108
+            # Windows timer resolution is typically limited to 15.6 ms by default.
```

Comment:
As discussed offline this seems only applicable to Python 3.10 and below; starting 3.11 the built-in `sleep` has a much better resolution.

Reply:
This could also explain why we did not catch it in the CI, as we only test PY312 on Windows due to budget constraints...

Reply:
I changed the comment as discussed offline.